### DISCUSSION TITLE
minDepth and ensureDepth

### DISCUSSION THREAD

**Discussion by 60pfennig** [payloadcms/payload#4744] [Feature Requests & Ideas]
Hey all,
I've found working with the API as well as the generated types a bit tedious. The problem in the uncertainty about the depth parameter. This results in super long type checking and error handling exspecially if you are working with multiple nested relationships. 
I also have read in Discord that others have the same problem ( [discord](https://discordapp.com/channels/967097582721572934/1163869339410239488 ) )

Thinking a bit about it I came to some possible solutions to tackle this problem:
- Introduce minDepth attribute for collections. This ensures a minimal depth getting returned by the API and so the types doesn't need to be an intersection type of string | T to some degree. Drawback is that the depth parameter of the Rest/Local API isn't working as expected because it can be overriden
- Providing a utility function which vaildates an object with given depth level and returns a casted type or undefined / error. This would be propably too complex for the Typescript type narrowing but maybe its possible to generate different depth types (Post.Depth1, Post.Depth2, ....) with the payload generate:types command.

Are there plans for addressing this problem? What are your thoughts? Am I missing something that already solves this?

**Comment by skysarwer** [comment]
I would love to see this. Can share the same sentiments

**Comment by tyteen4a03** [comment]
Related to https://github.com/payloadcms/payload/discussions/6161
